src/expression.rs

```diff
@@ -674,5 +692,29 @@ mod tests {
             prop_assert_eq!(x == y, h_x == h_y);
         }
 
+        #[test]
+        fn reals_are_real(x in any::<f64>()) {
+            prop_assert_eq!(Expression::Number(real!(x)).to_real(), Ok(x))
+        }
+
+        #[test]
+        fn some_nums_are_real(re in any::<f64>(), im in any::<f64>()) {
+            let result = Expression::Number(Complex64{re, im}).to_real();
+            if _is_small(im) {
+                prop_assert_eq!(result, Ok(re))
+            } else {
+                prop_assert_eq!(result, Err(EvaluationError::NumberNotReal))
+            }
+        }
+
+        #[test]
+        fn no_other_exps_are_real(expr in arb_expr().prop_filter("Not numbers", |e| match e {
+            Expression::Number(_) => false,
+            _ => true,
+        }
+            )) {
+            prop_assert_eq!(expr.to_real(), Err(EvaluationError::NotANumber))
+        }
+
```

Comment:
I'm not super familiar with the way `prop_test` picks values, but should there maybe be a concrete test with specifically `0.0` or something that might be commonly used in a Quil program?

Reply:
Thanks! Reminds me that I should probably handle π as well.

Reply:
yum.. 🥧